unity-renderer/Assets/DCLPlugins/BuilderInWorld/Publisher/ProjectPublishHUD/Scripts/Projects/PublishProjectDetailView.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace DCL.Builder
+{
+    public interface IPublishProjectDetailView
+    {
+        /// <summary>
+        /// If the publish is canceled this action will be called
+        /// </summary>
+        event Action OnCancel;
+
+        /// <summary>
+        /// If the publish button is pressed this action will be called
+        /// </summary>
+        event Action<PublishInfo> OnPublishButtonPressed;
+
+        /// <summary>
+        /// If the rotation of the project is changed, this event will be fired
+        /// </summary>
+        event Action<PublishInfo.ProjectRotation>  OnProjectRotateChange;
+
+        /// <summary>
+        /// Set the project to publish
+        /// </summary>
+        /// <param name="scene"></param>
+        void SetProjectToPublish(IBuilderScene scene);
+
+        /// <summary>
+        /// This will show the detail modal 
+        /// </summary>
+        void Show();
+
+        /// <summary>
+        /// This will hide the detail modal
+        /// </summary>
+        void Hide();
+
+        /// <summary>
+        /// Dispose the view
+        /// </summary>
+        void Dispose();
+    }
+
+    public class PublishProjectDetailView : BaseComponentView, IPublishProjectDetailView
+    {
+        private const string SCREENSHOT_TEXT =  @"{0} parcel = {1}x{2}m";
+
+        public event Action<PublishInfo.ProjectRotation> OnProjectRotateChange;
+        public event Action OnCancel;
+        public event Action<PublishInfo> OnPublishButtonPressed;
+
+        [Header("Map Renderer")]
+        public PublishMapView mapView;
+
+        [SerializeField] internal Button cancelButton;
+        [SerializeField] internal Button publishButton;
+
+        [SerializeField] internal Button rotateLeftButton;
+        [SerializeField] internal Button rotateRightButton;
```

Comment:
Why don't we use `ButtonComponentViews` here? In this way we can use the buttons implemented in our UI Components library.

Reply:
Ok, only for taking into account when you have to change this part: The use of a `Button` or a `BaseComponentView` is the same in terms of "overkill", both are used in the same way (you will subscribe to its `onClick` event), but the advantage of using a button from our UI library is that it will ensure to have a visual design (from the base prefab) centralized and coherent with the rest of our UIs.